src/core/datastore/deltaset.js

```diff
@@ -0,0 +1,159 @@
+import isArray from 'lodash/isArray';
+import { format } from 'url';
+import { Query } from '../query';
+import { KinveyRequest, RequestMethod, AuthType } from '../request';
+import { Client } from '../client';
+import { KinveyError } from '../errors';
+import { repositoryProvider } from './repositories';
+import { buildCollectionUrl } from './repositories/utils';
+import {
+  stripTagFromCollectionName,
+  generateEntityId,
+  queryCacheCollectionName,
+  xKivneyRequestStartHeader
+} from './utils';
+
+function getDeltaSetQuery(collectionNameWithTag, query) {
+  const serializedQuery = query ? query.toString() : '';
+
+  return repositoryProvider.getOfflineRepository()
+    .then((offlineRepo) => {
+      const queryCacheQuery = new Query()
+        .equalTo('collectionName', collectionNameWithTag)
+        .and()
+        .equalTo('query', serializedQuery);
+      return offlineRepo.read(queryCacheCollectionName, queryCacheQuery)
+        .then((deltaSetQueryDocs = []) => {
+          if (deltaSetQueryDocs.length > 0) {
+            return deltaSetQueryDocs[0];
+          }
+
+          return {
+            _id: generateEntityId(),
+            collectionName: collectionNameWithTag,
+            query: serializedQuery
+          };
+        });
+    });
+}
+
+function updateDeltaSetQuery(deltaSetQuery, response) {
+  return repositoryProvider.getOfflineRepository()
+    .then((offlineRepo) => {
+      deltaSetQuery.lastRequest = response.headers.get(xKivneyRequestStartHeader);
+      return offlineRepo.update(queryCacheCollectionName, deltaSetQuery);
+    });
+}
+
+function readDocs(collectionNameWithTag, query) {
+  return repositoryProvider.getOfflineRepository()
+    .then((offlineRepo) => {
+      return offlineRepo.read(collectionNameWithTag, query);
+    });
+}
+
+function deleteDocs(collectionNameWithTag, deleted = []) {
```

Comment:
This method signature is confusing. Is the purpose of the method to `delete` the entities that are passed in as an array parameter? If so, it's clearer to name that variable something like `entitiesToDelete` rather than `deleted`.